posts/ethereum-development-tips.md

```diff
@@ -0,0 +1,254 @@
+            uint256 blockValue = uint256(block.blockhash(block.number-1));
+
+            if (lastHash == blockValue) {
+                revert();
+            }
+
+            lastHash = blockValue;
+            uint256 coinFlip = blockValue / FACTOR;
+            bool side = coinFlip == 1 ? true : false;
+
+            if (side == _guess) {
+                consecutiveWins++;
+                return  true;
+            } else {
+                consecutiveWins = 0;
+                return  false;
+            }
+        }
+    }
+
+Be very cautious when you call other contracts, because they could hide data to execute, or use the call stack limit depth or gas to halt the execution where they want it to.
+
+However, do not be fooled thinking the risks can only come from malicious users, since Solidity may behave in ways you would not expect: overflow and underflow errors are legion in Solidity, nonexistent float type means your division will always be rounded...
+
+There is a special function that exists in a contract to address the need for floating point. It does not have a name, and will be executed when no other appropriate function is found. It is called the fallback function, and can be used for quite a few interesting hacking tricks. However as we found out, you can not close it up completely, “just to be sure”, as you need to have a payable fallback function if you want it to receive ETH from another contract!
+
+    function() public payable { //Fairly open now!
+
+    }
+
+Security in Solidity, as in every language, is a difficult system of checks and balances between features and protection.
+
+Of course we had to link the official solidity docs, they are well documented and updated regularly:
+
+[https://solidity.readthedocs.io/en/v0.4.24/](https://solidity.readthedocs.io/en/v0.4.24/)
+
+A classic game of increasingly difficult solidity hacking challenges. It gets extremely tough!
+
+[https://ethernaut.zeppelin.solutions/](https://ethernaut.zeppelin.solutions/)
+
+A coding contest that highlights a lot of mistakes or malicious attacks possible in Solidity. There’s lot of exemples to check out on the github!
+[https://medium.com/@weka/announcing-the-winners-of-the-first-underhanded-solidity-coding-contest-282563a87079](https://medium.com/@weka/announcing-the-winners-of-the-first-underhanded-solidity-coding-contest-282563a87079)
+
+A collection of "best practices" in the Ethereum development world by Consensys, a major actor in the Ethereum market.
+
+https://consensys.github.io/smart-contract-best-practices/  
+
+# Conclusion
+
+We hope this small introduction to Ethereum development has helped you. We also would like to sincerely thank the entire team of Ippon USA who hosted two French interns in Washington DC, and especially our tutor Chris Lumpkin, Dagmawi Mengistu, Priya Rajanna and Matt Reed our co-workers who welcomed and showed us a priceless glimpse of the American melting pot culture.
```

Comment:
our tutor Chris Lumpkin, Dagmawi Mengistu, Priya Rajanna and Matt Reed our co-workers -> confusing